### PR TITLE
Remove SunOS relics

### LINKED ISSUE #399: Remove Sun4 code
Doing a search with `sun4` and `PLAT_SUN4` reveals many hits.

### PR DESCRIPTION
**Pull request by miguelsousa**
Fixes #399

### CODE REVIEW

File: afdko/Tools/Programs/public/lib/api/supportenvironment.h [hunk 1]
```diff
@@ -285,7 +283,7 @@ that changes to environment.h be upward-compatible, as described above.
  */
 
 #ifndef PROTOTYPES
-#define PROTOTYPES	(OS!=os_sun)
+#define PROTOTYPES	(OS)
```
Comment (cjchapman) [inline afdko/Tools/Programs/public/lib/api/supportenvironment.h]: That's not how this is meant to be used -- see the comment block above this line for more information about it, but it doesn't seem to be used at all, so I think you should remove this and the comment block above it.

**Comment by miguelsousa** [comment]
@cjchapman another pass done

File: afdko/Tools/Programs/public/lib/api/atm.h [hunk 2]
```diff
@@ -62,16 +62,7 @@ buildch and other ATM modules (font parser).
...
 
-/*
- * Some of the PSENVIRONMENT configurations won't compile with ENABLE_ASSERT,
- * so we only turn it on for isp_sparc.  
- * This is enough to find machine-independent PSENVIRONMENT problems.
- */
-#if STAGE==DEVELOP && ((defined PSENVIRONMENT && !PSENVIRONMENT) || ISP==isp_sparc)
-#define ENABLE_ASSERT 1
-#else
 #define ENABLE_ASSERT 0
```
Comment (cjchapman) [inline afdko/Tools/Programs/public/lib/api/atm.h]: I think we want to enable asserts, and this should probably be:
```c
#if STAGE==DEVELOP && (defined PSENVIRONMENT && !PSENVIRONMENT)
#define ENABLE_ASSERT 1
#else
#define ENABLE_ASSERT 0
#endif
```

**Comment by miguelsousa** [comment]
@cjchapman done. OK to merge?

**Pull request merged**